crates/db_schema/src/utils.rs

```diff
@@ -26,11 +30,32 @@ pub fn fuzzy_search(q: &str) -> String {
   format!("%{}%", replaced)
 }
 
-pub fn limit_and_offset(page: Option<i64>, limit: Option<i64>) -> (i64, i64) {
-  let page = page.unwrap_or(1);
-  let limit = limit.unwrap_or(10);
+pub fn limit_and_offset(
```

Comment:
You can ignore the rest, this is the main change for limit and offset: it can now return a diesel error if the checks are out of bounds.

Reply:
Using NotFound error here will lead to a lot of confusion. Better to use QueryBuilderError with some details inside, or at least log a warning.